### PR TITLE
Making input_tester and expand immutable

### PR DESCRIPTION
**Pull request by chkeita**
Refactor input_tester and expand to make them immutable

### CODE REVIEW

File: src/agent/onefuzz-agent/src/tasks/fuzz/supervisor.rs [hunk 1]
```diff
@@ -183,23 +182,19 @@ async fn start_supervisor(
...
-    }
+        .task_id(&config.common.task_id)
+        .set_optional_ref(&config.tools, |expand, tools| expand.tools_dir(&tools.path))
+        .set_optional_ref(&config.target_exe, |expand, target_exe| {
+            expand.target_exe(target_exe)
+        })
+        .set_optional_ref(&config.supervisor_input_marker, |expand, input_marker| {
+            expand.input_marker(input_marker)
+        });
+    let expand = if let Some(target_options) = &config.target_options {
```
Comment (bmc-msft) [inline src/agent/onefuzz-agent/src/tasks/fuzz/supervisor.rs:193]: I see the use of `set_optional_ref`.  Can it be done here too?

Comment (chkeita) [reply src/agent/onefuzz-agent/src/tasks/fuzz/supervisor.rs]: I needed to update the declareation of `set_optional_ref` to include a life time paramater. without it i was getting this error 
```
   --> onefuzz-agent\src\tasks\fuzz\supervisor.rs:193:76
    |
193 |         .set_optional_ref(&config.target_options, |expand, target_options| expand.target_options(target_options))
    |                                                    ------  --------------  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `target_options` escapes the closure body here
    |                                                    |       |
    |                                                    |       `target_options` is a reference that is only valid in the closure body
    |                                                    `expand` declared here, outside of the closure body

```

**Review by bmc-msft — APPROVED** [review]

**Comment by bmc-msft** [comment]
Running integration tests

**Comment by bmc-msft** [comment]
This PR needs #505.

**Pull request merged**